### PR TITLE
Create convert_diffusers_sdxl_lora_to_webui.py

### PR DESCRIPTION
**Pull request by apolinario**
# What does this PR do?
Converts Diffusers SDXL LoRA to WebUIs format (AUTOMATIC1111, ComfyUI, SD.Next, etc) 

## Who can review?
@sayakpaul @younesbelkada

### CODE REVIEW

File: scripts/convert_diffusers_sdxl_lora_to_webui.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# Script for converting a HF Diffusers trained SDXL LoRAs (be it in the old, new or PEFT format)
```
Comment (sayakpaul) [inline scripts/convert_diffusers_sdxl_lora_to_webui.py:1]: I'd add some reference examples here too as I believe this is going to be used quite a lot by the community ❤️

File: scripts/convert_diffusers_sdxl_lora_to_webui.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+
+def convert_all_diffusers_to_peft(diffusers_dict):
+    try:
+        peft_dict = convert_state_dict_to_peft(diffusers_dict)
+    except Exception as e:
+        if str(e) == "Could not automatically infer state dict type":
+            peft_dict = convert_unet_state_dict_to_peft(diffusers_dict)
+        else:
+            raise
```
Comment (sayakpaul) [inline scripts/convert_diffusers_sdxl_lora_to_webui.py:20]: 🧠

File: scripts/convert_diffusers_sdxl_lora_to_webui.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+def convert_and_save(input_lora, output_lora=None):
+    if output_lora is None:
+        base_name = os.path.splitext(input_lora)[0]
+        output_lora = f"{base_name}_webui.safetensors"
+
+    diffusers_state_dict = load_file(input_lora)
+    peft_state_dict = convert_all_diffusers_to_peft(diffusers_state_dict)
+    kohya_state_dict = convert_peft_to_kohya(peft_state_dict)
+    save_file(kohya_state_dict, output_lora)
```
Comment (sayakpaul) [inline scripts/convert_diffusers_sdxl_lora_to_webui.py:60]: Do we wanna have an option for `push_to_hub`? That'd be very cool IMO.

**Review by sayakpaul — APPROVED** [review]
Lovely 😍

File: scripts/convert_diffusers_sdxl_lora_to_webui.py (hunk 1 above)
Comment (apolinario) [reply scripts/convert_diffusers_sdxl_lora_to_webui.py]: Done!

File: scripts/convert_diffusers_sdxl_lora_to_webui.py (hunk 3 above)
Comment (apolinario) [reply scripts/convert_diffusers_sdxl_lora_to_webui.py:30]: Love it! I'd merge it as is for now and add it in a next PR , we can also have other utils to PR a Kohya weight to existing diffusers trained repos and other cool utils like that

File: scripts/convert_diffusers_sdxl_lora_to_webui.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+# Get the output to work with WebUIs such as AUTOMATIC1111, ComfyUI, SD.Next and others.
+
+# Here are some diffusers training scripts and utils that you can use and then convert:
+# LoRA Ease - no code SDXL Dreambooth LoRA trainer: https://huggingface.co/spaces/multimodalart/lora-ease
+# Dreambooth Advanced Training Script - state of the art techniques such as pivotal tuning and prodigy optimizer:
+# - Script: https://github.com/huggingface/diffusers/blob/main/examples/advanced_diffusion_training/train_dreambooth_lora_sdxl_advanced.py
+# - Colab (only on Pro): https://colab.research.google.com/github/huggingface/notebooks/blob/main/diffusers/SDXL_Dreambooth_LoRA_advanced_example.ipynb
+# Canonical diffusers training scripts:
+# - Script: https://github.com/huggingface/diffusers/blob/main/examples/dreambooth/train_dreambooth_lora_sdxl.py
+# - Colab (runs on free tier): https://colab.research.google.com/github/huggingface/notebooks/blob/main/diffusers/SDXL_DreamBooth_LoRA_.ipynb
```
Comment (sayakpaul) [inline scripts/convert_diffusers_sdxl_lora_to_webui.py:12]: Can we also add a working example for the script itself? I think that might be helpful too!

**Review by younesbelkada — APPROVED** [review]
Thanks a lot @apolinario @sayakpaul for taking care of this!

**Review by sayakpaul — APPROVED** [review]
Slick!

File: scripts/convert_diffusers_sdxl_lora_to_webui.py (hunk 4 above)
Comment (apolinario) [reply scripts/convert_diffusers_sdxl_lora_to_webui.py:18]: ✅

**Pull request merged**